packages/kit/src/runtime/server/respond.js

```diff
@@ -54,9 +54,10 @@ const allowed_page_methods = new Set(['GET', 'HEAD', 'OPTIONS']);
  * @param {import('types').SSROptions} options
  * @param {import('@sveltejs/kit').SSRManifest} manifest
  * @param {import('types').SSRState} state
- * @returns {Promise<Response>}
+ * @param {{request: import('http').IncomingMessage, socket: import('stream').Duplex , head: Buffer}?} upgradeRequest
+ * @returns {Promise<void | Response>}
```

Comment:
I'm not that familiar with `UPGRADE`, but is it true that it sends no response? A quick peek at MDN makes it appear to me that it does send a response

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Upgrade#overview

Reply:
https://github.com/websockets/ws/blob/68614728c164eb55462ff12ba24cb30451d28ec6/lib/websocket-server.js#L379-L414

The response appears to be sent back through the socket itself.

Reply:
There does appear to be some variance in the way the response is sent though, for example [Bun](https://bun.sh/docs/api/websockets#start-a-websocket-server) does not send a response, but [Deno](https://github.com/denoland/deno/blob/ef7432c03f83ad9e9ca2812d0ab5653e87fa5259/ext/http/02_websocket.ts#L40-L107) does appear to return a standard HTTP response.

Socket.IO also [returns a response](https://github.com/socketio/socket.io/blob/91e1c8b3584054db6072046404a24e79a17c1367/packages/engine.io/lib/server.ts#L782-L822) but they refuse to just give you the resp object, and instead insist on binding directly to a Server so they can send it themselves.

If we look at the [Node.JS server upgrade event](https://nodejs.org/api/http.html#event-upgrade), they specify the data is sent back through the socket, and their on upgrade handler does not even support sending a standard response object

Reply:
I would love maintainer/community input here, its definitely a fun issue to try and solve for everyone :P

Reply:
It looks like Socket.IO may actually support the same interface
https://github.com/socketio/socket.io/blob/91e1c8b3584054db6072046404a24e79a17c1367/packages/engine.io/README.md?plain=1#L40-L58

Reply:
I have't had any luck getting this to work locally